File: Algorithms/ad-hoc/rectangles.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
    cin>>n;
    int i=1,cnt=0;

    int ans=0;
    while(i<=n)
    {
        ans=ans+(n/i);
        if(n>=i*i)
        cnt++;
        i++;
    }
    ans=ans+(cnt);
    ans=ans/2;

    cout<<ans;
    return 0;
}

File: Algorithms/ad-hoc/Rectangles/Solutions/Rectangles.java
import java.util.*;

class Rectangles{

	public static void main(String args[]){
		Scanner ob=new Scanner(System.in);
		int n=ob.nextInt();
		int i=1,cnt=0,ans=0;
		while(i<=n){
			ans+=(n/i);
			if(n>=i*i)
				cnt++;
			i++;
		}
		ans+=cnt;
		ans/=2;
		System.out.println(ans);
	}
}